src/runtime/refiner.ts

```diff
@@ -184,6 +184,11 @@ export class Refinement {
     return this.expression.toSQLExpression(codeGenerator);
   }
 
+  toProto(): object {
```

Comment:
You should not add code related to proto serialization to core runtime classes (such as this one). What we do with protos is just an implementation detail and likely a temporary one. We should work hard to contain it to proto serialization code. If you need to you can relax visibility of some members or access them without type safety in the proto serialization code.